src/components/Search/SearchInput.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useState, useEffect, useMemo, useRef } from 'react'
+import { Close } from 'decentraland-ui/dist/components/Close/Close'
+import TokenList from 'decentraland-gatsby/dist/utils/dom/TokenList'
+import useFormatMessage from 'decentraland-gatsby/dist/hooks/useFormatMessage'
+import { useLocation } from '@reach/router'
+
+import './SearchInput.css'
+import { navigate } from 'gatsby-plugin-intl'
+import locations from '../../modules/locations'
+
+export function handleSearch(textSearch:string) {
+  const newParams = new URLSearchParams(location.search)
+  if (textSearch) {
+    newParams.set('search', textSearch)
+    newParams.delete('page')
+    newParams.delete('order')
+  } else {
+    newParams.delete('search')
+    newParams.delete('page')
+  }
+
+  navigate(locations.proposals(newParams))
+}
+
+export default function SearchInput(props: React.HTMLAttributes<HTMLDivElement>) {
+  const l = useFormatMessage()
+  const location = useLocation()
+  const params = useMemo(() => new URLSearchParams(location.search), [location.search])
+  const search = useMemo(() => params.get('search') || '', [params])
```

Comment:
I don't think it's necessary to memo params and search as there is no expensive value to calculate that will change on every render.

Reply:
Ty! I'm removing the one for search, here and in other components.
Are you sure about the params with the `new URLSearchParams`? I think by memoizing we could avoid creating a new instance on every render

Reply:
You're right!

Reply:
@2fd do you have any input on this? we'd appreciate it! 😄